Fixes Execution error on budget ratio after migration

osarrat opened the pull request.

osarrat committed:
Merge pull request #1 from sigmah-dev/master

Pull from origin on 18/10/2016

osarrat committed:
Fixes Execution error on budget ratio after migration

Fixes #997 with a SQL function for budget ratio migration instead of an external Java tool

ghost commented:
@osarrat great work! If I understand well, that could be transformed to a flyway migration in an eyesight. The migration will just:
1. Create all required procedures (= your script)
2. call `migrate_budget_fields()` (one line to add to your script)
3. drop all procedures created at step 1. just to keep the database clean: those procedure are no longer needed once the migration is done so why keep those? (three line to add)

Would it be OK to you? It will still remain a SQL script that anybody (production guys...) could call.

ghost commented:
Also, still about line-endings, I might be wrong. Can you try this:

``` bash
git config --global core.autocrlf true
```

That's the [proper way to deal with line-endings for Windows user according to github](https://help.github.com/articles/dealing-with-line-endings/#platform-windows).

osarrat committed:
Add bigint management + call to MigrateBudgetField

issue #997 (Execution error on budget ratio after migration with
dedicated Java tool)

osarrat commented:
Hi @numero-six 
I have updated the script to that it does not only create the procedure, but it also calls it and drop it. I have also fixed an issue raised while migrating a large database with bigint ids. Finally, I have let intentionnally the other procedures, because they are useful and can be used for example to import data through SQL.

Regarding line-endings, I have upated my repo and [the wiki](http://wiki.sigmah.org/doku.php?id=contributorguide:preparebuildenvironment#step_2get_a_local_copy_of_the_source) accordingly. Thanks !

osarrat committed:
Fixes history issue on migrate budget fields

And move the SQL script to Flyway dir.
issue #997
